lib/apytypes/_array_functions.py

```diff
@@ -411,9 +411,467 @@ def expand_dims(a, axis):
+
+    Parameters
+    ----------
+    a : :class:`APyFloatArray` or :class:`APyFixedArray`
+        The shape and data-type of a define these same attributes of the returned array.
+    int_bits : int, optional
+        Number of integer bits.
+    frac_bits : int, optional
+        Number of fractional bits.
+    exp_bits : int, optional
+        Number of exponential bits.
+    mantissa_bits : int, optional
+        Number of mantissa bits.
+
+    Returns
+    -------
+    result : :class:`APyFloatArray` or :class:`APyFixedArray`
+        The array initialized filled with ones.
+    """
+    try:
+        ones = a.ones
+    except AttributeError:
+        raise TypeError(f"Cannot make ones array of type {type(a)}")
+
+    from apytypes import APyFixedArray, APyFloatArray
+
+    if isinstance(a, APyFixedArray):
+        int_bits = int_bits or a.int_bits
+        frac_bits = frac_bits or a.frac_bits
+        return ones(shape=a.shape, int_bits=int_bits, frac_bits=frac_bits)
+
+    if isinstance(a, APyFloatArray):
+        exp_bits = exp_bits or a.exp_bits
+        mantissa_bits = mantissa_bits or a.man_bits
+        return ones(shape=a.shape, exp_bits=exp_bits, man_bits=mantissa_bits)
+
+    raise ValueError("Only 'fixed' and 'float' array_types are defined")
+
+
+def full_like(
+    a,
+    fill_value,
+    int_bits=None,
+    frac_bits=None,
+    exp_bits=None,
+    mantissa_bits=None,
+):
+    """
+    Return a full array with the same shape and type as a given array.
+
+    If fill_value is an int or float, you must specify the word lengths (int_bits, frac_bits or exp_bits, mantissa_bits).
+    If fill_value is an APyFloat or APyFixed, the array will use the provided word lengths if specified.
+    If no word lengths are specified, the resulting array will inherit the word lengths from fill_value.
+
+    Parameters
+    ----------
+    a : :class:`APyFloatArray` or :class:`APyFixedArray`
+        The shape and array type of the returned array.
+    fill_value : :class:`APyFloat` or :class:`APyFixed` or int or float
+        The value to fill the array with.
+    int_bits : int, optional
+        Number of integer bits for APyFixed.
+    frac_bits : int, optional
+        Number of fractional bits for APyFixed.
+    exp_bits : int, optional
+        Number of exponent bits for APyFloat.
+    mantissa_bits : int, optional
+        Number of mantissa bits for APyFloat.
+
+    Returns
+    -------
+    result : :class:`APyFloatArray` or :class:`APyFixedArray`
+        Array filled with fill_value, having the same shape and type as `a`.
+    """
+    try:
+        full = a.full
+    except AttributeError:
+        raise TypeError(f"Cannot make full array of type {type(a)}")
+
+    fill_value = _normalize_fill_value(
+        fill_value, int_bits, frac_bits, exp_bits, mantissa_bits
+    )
+    return full(a.shape, fill_value)
+
+
+def arange(
+    start,
+    stop=None,
+    step=None,
+    int_bits=None,
+    frac_bits=None,
+    exp_bits=None,
```

Comment:
This will not work in case one of them is zero. (For the case below it is less likely, but here it can easily be the case.)

`int_bits is not None` etc.

Reply:
Nice catch! Should be fixed in in the latest commit now.